docs/security.adoc

```diff
@@ -57,3 +57,50 @@ image:images/roleBasedStrategyManage.png[image]
 The read-only role can then be assigned to either anonymous or authenticated users, leaving the dedicated Swarm role for Swarm users only:
 
 image:images/roleBasedStrategyAssign.png[image]
+
+== Caveats and Troubleshooting
+
+A single Jenkins controller currenly supports a single selection from a multitude of possible authentication providers.
+If your Jenkins deployment outgrows locally-defined accounts and switches to an external database, such as LDAP, GitHub or Google authentication, you would have to define there not only team member accounts, but also some for your Swarm agents to use.
+
+=== GitHub Authentication plugin
+
+With "Github Authentication Plugin" selected as the active Security Realm implementation for a Jenkins controller, it is no longer possible to authenticate the Swarm client with an account defined only locally on the Jenkins controller with a password (since the local database was no longer selected to be used for authentication), although this account is otherwise known and manageable in Jenkins UI.
+
+As tracked in https://issues.jenkins.io/browse/JENKINS-63421[issue JENKINS-63421] currently the "GitHub Committer Authorization Strategy" delivered as an optional part of the https://plugins.jenkins.io/github-oauth/[GitHub Authentication plugin] (`github-oauth-plugin`) lacks a way to assign *Agent/...* permissions.
+Due to this, you would have to define a Matrix-based or Project-based Matrix strategy as detailed above, and use the account (and group/organization) names provided and confirmed by its Authentication part as your enabled Security Realm.
+Note this can forfeit some benefits compared to native strategy provided by the plugin: it is questionable whether the matrix setup can go into such detail as `github-oauth-plugin` natively can -- e.g. considering PR authors, collaborators, contributors... as special groups).
+
+==== Primary setup
+
+Swarm clients can connect to the Jenkins controller using a GitHub account, and passing the token string with one of the `-password*` CLI options for the Swarm agent JAR startup.
+
+On the Jenkins controller side, the active authorization strategy has to be switched from "GitHub Committer Authorization Strategy" to a manually made somewhat equivalent Matrix. 
+It would be statically listing names of your project's GitHub organizations and teams of admin and other accounts as groups with specific rights.
+Be sure to not lock yourself out in the process -- perhaps adding an entry for your GitHub user name explicitly, assigning all privileges to it, and applying the change as the first step.
```

Comment:
The "be sure to not lock yourself out" sounds like a warning to me. Would be a good use case for an AsciiDoc "WARNING:" admonition.

Reply:
Gave this a shot in subsequent commits